Zep plugin

obadakhalili opened the pull request:
# Plugin Submission Form

## 1. Submission Type
- [x] New plugin submission
- [ ] Version update for existing plugin

## 2. Description

A memory plugin for Dify using Zep.

From [the docs](https://help.getzep.com/concepts#:~:text=Zep%20is%20a%20memory%20layer%20for%20AI%20assistants%20and%20agents%20that%20continuously%20learns%20from%20user%20interactions%20and%20changing%20business%20data.%20Zep%20ensures%20that%20your%20Agent%20has%20a%20complete%20and%20holistic%20view%20of%20the%20user%2C%20enabling%20you%20to%20build%20more%20personalized%20and%20accurate%20user%20experiences.):
> Zep is a memory layer for AI assistants and agents that continuously learns from user interactions and changing business data. Zep ensures that your Agent has a complete and holistic view of the user, enabling you to build more personalized and accurate user experiences.

GitHub: https://github.com/obadakhalili/dify-zep-plugin

## 3. Checklist
- [x] I have read and followed the Publish to Dify Marketplace guidelines
- [x] I have read and comply with the Plugin Developer Agreement
- [x] I confirm my plugin works properly on both Dify Community Edition and Cloud Version
- [x] I confirm my plugin has been thoroughly tested for completeness and functionality
- [x] My plugin brings new value to Dify

## 4. Documentation Checklist
Please confirm that your plugin README includes all necessary information:
- [x] Step-by-step setup instructions
- [x] Detailed usage instructions
- [x] All required APIs and credentials are clearly listed
- [x] Connection requirements and configuration details

## 5. Privacy Protection Information
Based on Dify Plugin Privacy Protection [Guidelines](https://docs.dify.ai/plugins/publish-plugins/publish-to-dify-marketplace/plugin-privacy-protection-guidelines):

### Data Collection

The plugin stores the user's conversations history.

### Privacy Policy
- [x] I confirm that I have prepared and included a privacy policy in my plugin package based on the Plugin Privacy Protection Guidelines

LogicOber commented:
@obadakhalili 

Thank you for your contribution. 💙

It would be best to add a URL field in the provider's YAML file. This way, users can be guided to obtain the relevant credential information. Ideally, this URL should point to a documentation page.

For instance:
```
  password:
    type: secret-input
    required: true
    label:
      en_US: Password
      zh_Hans: 密码
      pt_BR: Senha
    placeholder:
      en_US: Please enter your Confluence password
      zh_Hans: 请输入你的 Confluence 密码
    help:
      en_US: Enter your Confluence password or contact your administrator to get access
      zh_Hans: 请输入你的 Confluence 密码，或者联系管理员获取
    url: https://www.confluence.com/
```

LogicOber reviewed (changes requested):
@obadakhalili 

Thank you so much for your pull request on GitHub! 😊 Your contributions are greatly appreciated!

Please ensure that:

 All content in PRIVACY.md, README, and code remains in English only
Looking forward to your updates! 💯

obadakhalili commented:
done @LogicOber.

LogicOber reviewed (changes requested):
Please modify the file to ensure that `difypkg` can be correctly installed on the server and that the API KEY can be properly entered. Currently, there is no response after entering the API KEY.

obadakhalili commented:
@LogicOber, this is actually outside my control. it is a problem [from Dify's end](https://github.com/langgenius/dify/issues/16769). I would appreciate it if you could help with this,

LogicOber commented:
> @LogicOber, this is actually outside my control. it is a problem [from Dify's end](https://github.com/langgenius/dify/issues/16769). I would appreciate it if you could help with this,

Got it, I will reply to you as soon as possible.

LogicOber commented:
Hello, @obadakhalili this kind of issue usually happens because the plugin you developed isn’t running properly. Lambda won’t return any error messages when it fails — it just hangs at the gateway. This doesn’t seem to be on our side, so you might want to check it on your end.

It would be great if you could provide a detailed README file and demonstrate that your plugin runs correctly after being installed via Dify packaging.

obadakhalili commented:
sorry for the late reply.

> by the way, where's your requirements.txt?

I think you just pinpointed the issue!

nothing with the code was running incorrectly. I thought the issue was akin to [an issue mem0 plugin faced](https://github.com/langgenius/dify/issues/16769#issuecomment-2757471208) because the client package they used wrote files to disk, but it wasn't solved for me when I replaced the zep's client code with direct HTTP requests.

however, the package worked instantly when I added the `requirements.txt` file, which might make sense if this is a project that needs to be set up on a remote server as a lambda service. was this hinted at somewhere in the documentation, or is the oversight on my side?

@LogicOber I updated the package to include the requirements.txt.

LogicOber commented:
Glad to hear that ~ Hope your plugin becomes popular on the marketplace ~ Thank you for your contribution! 💙

obadakhalili commented:
thanks for your help! ❤️
